### PR TITLE
support RTB yaw to direction of motion: APM Side

### PR DESCRIPTION
**Pull request by CraftyJax**
allow POSVEL commands from planck_ctrl to carry yaw commands as well
part of resolution to https://planckaero.atlassian.net/browse/ACE-438

### CODE REVIEW

File: ArduCopter/mode_plancktracking.cpp [hunk 1]
```diff
@@ -174,7 +176,8 @@ void ModePlanckTracking::run() {
                           pos_cmd.z = new_alt_cm;
                       }
                   }
-                  ModeGuided::set_destination_posvel(pos_cmd,vel_cmd);
+
+                  ModeGuided::set_destination_posvel(pos_cmd,vel_cmd,true,yaw_cmd);
```
Comment (almatthew) [inline ArduCopter/mode_plancktracking.cpp]: Should we also add support for a yaw rate in this command?

File: libraries/AC_Planck/AC_Planck.cpp [hunk 2]
```diff
@@ -253,11 +253,12 @@ bool AC_Planck::get_position_cmd(Location &loc)
 }
 
 //Get a position, velocity cmd
-bool AC_Planck::get_posvel_cmd(Location &loc, Vector3f &vel_cms)
+bool AC_Planck::get_posvel_cmd(Location &loc, Vector3f &vel_cms, float &yaw_cd)
 {
   if(!_cmd.is_new) return false;
   loc = _cmd.pos;
   vel_cms = _cmd.vel_cms;
+  yaw_cd = _cmd.att_cd.z;
```
Comment (almatthew) [inline libraries/AC_Planck/AC_Planck.cpp:261]: `att_cd.z`?

**Review by almatthew — CHANGES_REQUESTED** [review]

File: ArduCopter/mode_plancktracking.cpp (hunk 1 above)
Comment (AubreyC) [reply ArduCopter/mode_plancktracking.cpp]: Kind of the same thing, but should the `yaw_cmd` be optional ? (probably achieved with `yaw_rate = 0` though)

File: ArduCopter/mode_plancktracking.cpp [hunk 3]
```diff
@@ -177,7 +179,12 @@ void ModePlanckTracking::run() {
                       }
                   }
 
-                  ModeGuided::set_destination_posvel(pos_cmd,vel_cmd,true,yaw_cmd);
+                  if(is_yaw_rate)
+                  {
+                    yaw_rate_cmd = yaw_cmd;
+                  }
+
```
Comment (almatthew) [inline ArduCopter/mode_plancktracking.cpp]: Extra whitespace

File: ArduCopter/mode_plancktracking.cpp [hunk 4]
```diff
@@ -145,11 +145,13 @@ void ModePlanckTracking::run() {
               Location loc_cmd;
               Vector3f vel_cmd;
               float yaw_cmd;
-
+              bool is_yaw_rate;
+              float yaw_rate_cmd = 0;
               bool good_cmd = copter.planck_interface.get_posvel_cmd(
```
Comment (almatthew) [inline ArduCopter/mode_plancktracking.cpp:151]: Missing a line

**Review by almatthew — CHANGES_REQUESTED** [review]

**Review by almatthew — APPROVED** [review]

**Pull request merged**